pkgs/fc/agent/fc/util/enc.py

```diff
@@ -294,9 +306,14 @@ def update_enc(log, tmpdir, enc_path):
         os_release_file=Path("/etc/os-release"),
         state_version_file=STATE_VERSION_FILE,
     )
-    enc = load_enc(log, enc_path)
-    update_inventory(log, enc)
+    # We need directory connection info from enc.json first to retrieve the
+    # new enc.json and other inventory files.
+    pre_enc = load_enc(log, enc_path)
+    update_inventory(log, pre_enc)
     # Reload to pick up possibly updated enc.json.
     enc = load_enc(log, enc_path)
     update_enc_nixos_config(log, enc, enc_path)
     write_system_state(log)
+    # Placed at the end because failures here are the least problematic ones.
+    # We can continue with missing release info data.
+    write_release_info(log, enc)
```

Comment:
And this is the change that prevents us from persisting the release metadate before we know it can be instantiated correctly. Also good.